src/babel-serializer/geometry/off.hh

```diff
@@ -1,3 +1,43 @@
 #pragma once
 
-// TODO
+#include <cstddef> // std::byte
+
+#include <clean-core/span.hh>
+#include <clean-core/typedefs.hh>
+#include <clean-core/vector.hh>
+
+#include <typed-geometry/tg-lean.hh>
+
+#include <babel-serializer/errors.hh>
+
+namespace babel::off
+{
+struct read_config
+{
+    bool has_colors = false;
+    bool has_normals = false;
+    bool has_tex_coords = false;
+    int dimension = 3;
```

Comment:
what is the dimension?

Reply:
ah if you mean this stella 4D stuff: that sounds incredibly niche :D

Reply:
Technically OFF the specification says that the dimension can be arbitrary, I believe allowing at most 4-dimensional points is a sensible default :)